src/Cohesity.Powershell/Scripts/Restore/Restore-CohesityFileV2.ps1

```diff
@@ -0,0 +1,405 @@
+                }
+                else  {
+
+                    $isDirectory = $false
+                }
+                $restoreParams.vmwareParams.recoverFileAndFolderParams.filesAndFolders += @{
+                    "absolutePath" = $file;
+                    "isDirectory" = $isDirectory
+                }
+            }
+            else  {
+
+                $searchParams = @{
+                    "fileParams" = @{
+                        "searchString"       = $file;
+                        "sourceEnvironments" = @(
+                            "kVMware"
+                        );
+                        "objectIds"          = @(
+                            $objectId
+                        )
+                    };
+                    "objectType" = "Files"
+                }
+                $url = "/v2/data-protect/search/indexed-objects"
+                $searchJson = $searchParams | ConvertTo-Json -Depth 100
+                $search = Invoke-RestApi -Method Post -Uri $url -Body $searchJson
+
+                $thisFile = $search.files | Where-Object {("{0}/{1}" -f $_.path, $_.name) -eq $file -or ("{0}/{1}/" -f $_.path, $_.name) -eq $file}
+                if(!$thisFile)  {
+
+                    $errorMsg = "file $file not found"
+                    Write-Output $errorMsg
+                    CSLog -Message $errorMsg
+
+                }
+                else  {
+                    if($file[-1] -eq '/')  {
+
+                        $isDirectory = $True
+                        $absolutePath = "{0}/{1}/" -f $thisFile[0].path, $thisFile[0].name
+                    }
+                    else  {
+
+                        $isDirectory = $false
+                        $absolutePath = "{0}/{1}" -f $thisFile[0].path, $thisFile[0].name
+                    }
+                    $restoreParams.vmwareParams.recoverFileAndFolderParams.filesAndFolders += @{
+                        "absolutePath" = $absolutePath;
+                        "isDirectory" = $isDirectory
+                    }
+                }
+            }
+        }
+
+        # perform restore
+        if($restoreParams.vmwareParams.recoverFileAndFolderParams.filesAndFolders.Count -gt 0)  {
+
+            $url = "/v2/data-protect/recoveries"
+            $restoreJson = $restoreParams | ConvertTo-Json -Depth 100
+            $restoreTask = Invoke-RestApi -Method Post -Uri $url -Body $restoreJson
+            $restoreTaskId = $restoreTask.id
+            if($wait)  {
+
+                Start-Sleep 3
+                while($restoreTask.status -eq "Running")  {
+
+                    Start-Sleep 5
+                    $url = "/v2/data-protect/recoveries/$($restoreTaskId)?includeTenants=true"
+                    $restoreTask = Invoke-RestApi -Method Get -Uri $url
+                    $restoreTask
+                }
+                if($restoreTask.status -eq 'Succeeded')  {
+
+                    Write-Output "Restore $($restoreTask.status)"
```

Comment:
To maintain consistency, give the entire restore task , instead of the status.

Reply:
ok